### DISCUSSION TITLE
Calling Azure Identity in middleware makes it think it is running client side

### DISCUSSION THREAD

**Discussion by tagoad** [vercel/next.js#34696] [Help]
So i am using Azure Key Vault, and accessing it works just fine from within the api endpoints i have created but I am setting up middleware to handle authentication of requests and when it runs instead of using Azure Identity's Nodeflow for authentication it keeps using its browserflow which errors out. What is about calling these functions within middleware that changes its behaviour to think its client side? Any pointers on how to fix this?

**Comment by balazsorban44** [comment]
Hi, this is because Middleware is run in the Edge Runtime, which is closer to the Browser environment in terms of APIs. (since browsers are following the standard Web APIs)

This is mentioned here:
https://nextjs.org/docs/api-reference/edge-runtime#unsupported-apis

We are working on making this clear.

But to note, your Middleware does **not** actually run in the browser.

**Reply by tagoad** [reply]
Thanks! Is there anyway to work around this? I would hate to program the authentication into each endpoint separately...

**Reply by balazsorban44** [reply]
Not sure what you mean by that. if you want to reuse code just abstract it away into a file and import it wherever you need it.

**Reply by tagoad** [reply]
Yes of course, but still not as clean and tidy as using it as middleware where there's a single call and import to the abstracted code
